src/java/org/apache/cassandra/db/rows/UnfilteredRowIteratorWithLowerBound.java

```diff
@@ -186,7 +193,7 @@ private static <V> ClusteringBound<V> createArtificialLowerBound(boolean isRever
     /**
      * @return the lower bound stored on the index entry for this partition, if available.
      */
-    private ClusteringBound<?> getPartitionIndexLowerBound()
+    private ClusteringBound<?> getKeyCacheLowerBound()
     {
         // NOTE: CASSANDRA-11206 removed the lookup against the key-cache as the IndexInfo objects are no longer
```

Comment:
Isn't this note invalid?

Also, I don't think we want to load the partition if `canUseMetadataLowerBound` is false -- we can just forego the bound in that case.

Reply:
I believe this comment is not correct (row indexes can also reside on the heap, if they are small enough, and the `rowIndexEntry.indexOnHeap()` call below checks this).